docs/preview/03-Features/01-Azure/01-service-bus.md

```diff
@@ -0,0 +1,291 @@
+
+### Register the Arcus message pump
+There exists two type of Azure Service bus *message pumps*: for queues and for topic subscriptions. During the registration of the pump in the application services, the type of authentication mechanism can be configured.
+
+> 🎖️ Use the [`ManagedIdentityCredential`](https://learn.microsoft.com/en-us/dotnet/api/azure.identity.managedidentitycredential) where possible.
+
+```csharp
+using Microsoft.Extensions.DependencyInjection;
+
+Host.CreateDefaultBuilder()
+    .ConfigureServices(services =>
+    {
+        // Register the message pump via a `TokenCredential` implementation.
+        // 🔗 See Microsoft's guidance on Azure authentication: 
+        // https://learn.microsoft.com/en-us/dotnet/api/overview/azure/identity-readme
+        services.AddServiceBusQueueMessagePump(
+            "<queue-name>",
+            "<fully-qualified-namespace>",
+            new ManagedIdentityCredential());
+
+
+        // Register the message pump via a custom implementation factory, returning a `ServiceBusClient`.
+        // 🔗 See Microsoft's guidance on Azure core extensions to re-use clients in your application:
+        // https://learn.microsoft.com/en-us/dotnet/api/overview/azure/microsoft.extensions.azure-readme
+        services.AddServiceBusTopicMessagePump(
+            "<topic-name>",
+            "<subscription-name>",
+            "<fully-qualified-namespace>",
+            (IServiceProvider services) =>
+            {
+                return services.GetRequiredService<ServiceBusClient>();
+            });
+    });
+```
+
+### Register your message handlers on message pump
+Your custom *message handlers* can be registered on a *message pump* registration. Both the type of the handler and type of of messages it can process, is required.
+
+```csharp
+using Microsoft.Extensions.DependencyInjection;
+
+Host.CreateDefaultBuilder()
+    .ConfigureServices(services =>
+    {
+        services.AddServiceBusTopicMessagePump(...)
+                .WithServiceBusMessageHandler<OrderMessageHandler, Order>()
+                .WithServiceBusMessageHandler<OrderV2MessageHandler, OrderV2>();
+    });
+```
+
+> **⚠️ Considerations:**
+> * The order in which the *message handlers* are registered matters when a message is routed.
+> * Only the first matching *message handler* will be used to route the message, even when multiple match.
+> * Multiple *message handlers* with the same (message) type can registered, but they need to distinguish themselves with [routing options](#message-handler-routing-customization).
+
+## Customization
+Due to the wide range of situations within messaging solutions, Arcus Messaging supports highly customizable message pump/handler registrations.
+
+### Message pump customization
```

Comment:
customization or rather configuration ?

Reply:
In a way, all options here is to 'customize' additional functionality, no?

Reply:
In my head, customization goes a bit further than configuration, but I'm ok with it.